src/linker/Linker.Analysis/README.md

```diff
@@ -0,0 +1,61 @@
+# Linker analyzer
+
+We have been prototyping a callgraph-based analyzer for the linker, that reports "linker-unsafe" usage in the IL reachable from some set of entry points (Main, or all public surface area). It simulates attributes that influence how unsafe patterns get reported.
+
+# Instructions
+
+To give it a try:
+
+## Clone and build
+```
+> git clone https://github.com/vitek-karas/linker -b Analysis
+> cd linker
+linker> ./eng/dotnet.{sh/ps1} build illink.sln
+```
+
+## Run
+
+Run the linker specifying the desired set of roots, optimizations, etc. A summary of the analysis will be written to stdout, and the "callstacks" will be written to `trimanalysis.json` in the linker output directory.
+For example: 
+
+```
+linker> dotnet ./src/ILLink.Tasks/bin/Debug/netcoreapp2.0/illink.dll -a console -c link -u link -d testapps/console/out/ --skip-unresolved true --ignore-descriptors true --verbose --dump-dependencies --used-attrs-only true -out testapps/console/linked --analyze-for-trim
+```
+
+Here, the command-line flags have the following meaning.
+```
+-a console:                    Roots console.dll. The linker will see that it is an application, and root only the entry point (not all public APIs)
+-c link -u link:               Tells the linker to trim all application (-u) and framework (-c) libraries unless otherwise specified.
+-d testapps/console/out:       Tells the linker where to find the dependencies of the application.
+--skip-unresolved:             Continue when encountering unresolved dependencies.
+--ignore-descriptors true:     Don't root types/members specified in embedded .xml descriptors. Use this option to see only what is kept by the linker's analysis.
```

Comment:
Also not sure we should use this - it's not how people will run the linker... right?

Reply:
I added this because I think there's a desire to move away from root descriptors since they kick in too unconditionally. IIRC we shipped with descriptors in 3.0 but Stephen has eliminated some of them. So I thought it was best to ignore them to make the analysis more complete.

Reply:
I just think that if we don't include them, we kind of intentionally break stuff - they were added for a reason (mostly) :-)
Not super important though - it will not really affect analysis that much.

Reply:
Yeah, that's a good point... especially the one for SPC. I'll take it out.